pilot/pkg/serviceregistry/aggregate/controller_test.go

```diff
@@ -534,3 +540,109 @@ func TestGetRegistries(t *testing.T) {
 		}
 	}
 }
+
+func newFakeController(clusterID string) (*kubecontroller.Controller, *fake.Clientset, func()) {
+	clientSet := fake.NewSimpleClientset()
+	c := kubecontroller.NewController(clientSet, kubecontroller.Options{
+		WatchedNamespace: "", // tests create resources in multiple ns
+		DomainSuffix:     "cluster.local",
+		ClusterID:        clusterID,
+	})
+
+	stop := make(chan struct{})
+
+	go c.Run(stop)
+
+	return c, clientSet, func() { close(stop) }
+}
+
+func buildMockKubeControllerForMultiCluster() (*Controller, func()) {
+	cluster1 := "cluster1"
+	cluster2 := "cluster2"
+	controller1, client1, cancel1 := newFakeController(cluster1)
+	controller2, client2, cancel2 := newFakeController(cluster2)
+
+	registry1 := Registry{
+		Name:             serviceregistry.ServiceRegistry("mockKubernetes1"),
+		ClusterID:        cluster1,
+		ServiceDiscovery: controller1,
+		Controller:       controller1,
+	}
+
+	registry2 := Registry{
+		Name:             serviceregistry.ServiceRegistry("mockAdapter2"),
+		ClusterID:        cluster2,
+		ServiceDiscovery: controller2,
+		Controller:       controller2,
+	}
+
+	ctls := NewController()
+	ctls.AddRegistry(registry1)
+	ctls.AddRegistry(registry2)
+
+	client1.CoreV1().Pods("test1").Create(&corev1.Pod{
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "test1",
+			Labels: map[string]string{
+				"app": "cluster1",
+			},
+		},
+		Spec: corev1.PodSpec{},
+		Status: corev1.PodStatus{
+			PodIP: "192.168.0.10",
+			Phase: corev1.PodRunning,
+		},
+	})
+
+	client2.CoreV1().Pods("test2").Create(&corev1.Pod{
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      "pod2",
+			Namespace: "test2",
+			Labels: map[string]string{
+				"app": "cluster2",
+			},
+		},
+		Spec: corev1.PodSpec{},
+		Status: corev1.PodStatus{
+			PodIP: "192.168.0.10",
+			Phase: corev1.PodRunning,
+		},
+	})
+
+	return ctls, func() {
+		cancel1()
+		cancel2()
+	}
+}
+
+func TestGetProxyWorkloadLabels(t *testing.T) {
```

Comment:
The change you're trying to test is in the discovery server. How does this test your change?

Reply:
The core modify is in aggregate